spec/03-plugins/36-request-transformer/02-access_spec.lua

```diff
@@ -2227,4 +2227,427 @@ describe("Plugin: request-transformer(access) [#" .. strategy .. "]", function()
     end)
   end)
 end)
+
+describe("Plugin: request-transformer(access) [#" .. strategy .. "] untrusted_lua=off", function()
+  local client
+
+  lazy_setup(function()
+    local bp = helpers.get_db_utils(strategy, {
+      "routes",
+      "services",
+      "plugins",
+    })
+
+    local route1 = bp.routes:insert({
+      hosts = { "testLuaOff.test" }
+    })
+
+    local route2 = bp.routes:insert({
+      hosts = { "testLuaOff2.test" }
+    })
+
+    bp.plugins:insert {
+      route = { id = route1.id },
+      name = "request-transformer",
+      config = {
+        add = {
+          headers = {"x-added:some_value"},
+        }
+      }
+    }
+
+    bp.plugins:insert {
+      route = { id = route2.id },
+      name = "request-transformer",
+      config = {
+        add = {
+          headers = {"x-added:$(require('inspect')(string.sub(query_params.user, 1, 3)))"},
+        }
+      }
+    }
+
+    assert(helpers.start_kong({
+      database = strategy,
+      untrusted_lua = "off",
+      plugins = "bundled, request-transformer",
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+    }))
+  end)
+
+  lazy_teardown(function()
+    helpers.stop_kong()
+  end)
+
+  before_each(function()
+    client = helpers.proxy_client()
+  end)
+
+  after_each(function()
+    if client then client:close() end
+  end)
+
+
+  it("correctly handles plain text value", function()
+    local r = assert(client:send {
+      method = "GET",
+      path = "/",
+      headers = {
+        host = "testLuaOff.test",
+        ["Content-Type"] = "application/x-www-form-urlencoded",
+      }
+    })
+    assert.response(r).has.status(200)
+    local value = assert.request(r).has.header("x-added")
+    assert.equals("some_value", value)
+  end)
+
+  it("does not render lua expressions", function()
+    local r = assert(client:send {
+      method = "GET",
+      path = "/",
+      query = {
+        user = "foo"
+      },
+      headers = {
+        host = "testLuaOff2.test",
+        ["Content-Type"] = "application/x-www-form-urlencoded",
+      }
+    })
+    assert.response(r).has.status(200)
+    local value = assert.request(r).has.header("x-added")
+    assert.equals("$(require('inspect')(string.sub(query_params.user, 1, 3)))", value)
```

Comment:
I am not sure this (rendering the template without any evaluation) is equivalent to what we do in the serverless function plugin(s). I think on those we [always call the sandbox](https://github.com/Kong/kong/blob/master/kong/plugins/pre-function/_handler.lua#L22), and then the sandbox [throws an error](https://github.com/Kong/kong/blob/master/spec/01-unit/20-sandbox_spec.lua#L111) like with the contents of `sandbox.configuration.err_msg`:

```suggestion
    assert.response(r).has.status(500)
    assert.request(r).not.has.header("x-added")
    -- and perhaps "the error log now shows the contents of sandbox.configuration.err_msg somewhere
```

In other words - ideally the plugin code should not "look" at the config at all. Instead it should always run the sandbox, passing the config options to it. I don't know how feasible this is, given that we run the code using penlight's template thingie.

Reply:
discussed and decided to "detect" lua inline expressions and throw error instead of returning non-parsed template